### ISSUE TITLE
RGroupDecomposition: bad handling of spiro-substitution

### ISSUE THREAD

**Issue by greglandrum** [rdkit/rdkit#3896]
**Describe the bug**
Here's a reproducible from 2020.09.5:
```
In [3]: cores = [Chem.MolFromSmiles("C1CCC1[*:3]")]
   ...: smi = ["C1CCC1O", "C1CCC1F", "C1CCC12COC2",]
   ...: mols = [Chem.MolFromSmiles(s) for s in smi]
   ...: groups,unmatched = rdRGroupDecomposition.RGroupDecompose(cores,mols,asSmiles=True,asRows=True)

In [4]: groups
Out[4]: 
[{'Core': 'C1C([*:1])CC1[*:3]', 'R1': 'O[*:1].[H][*:1]'},
 {'Core': 'C1C([*:1])CC1[*:3]', 'R1': 'F[*:1].[H][*:1]'},
 {'Core': 'C1C([*:1])CC1[*:3]', 'R1': 'C(OC[*:1])[*:1]'}]
```

The same thing happens with master + the changes from #3767, but there the extra H atoms are missing (which is good).

Here are those results in pictures:
![image](https://user-images.githubusercontent.com/540511/110332575-c9548380-8020-11eb-9982-0e530798dc74.png)

**Expected behavior**
Given that the input specified that the R group is R3 and that R1 doesn't appear anywhere, I believe the output should be:
```
[{'Core': 'C1CCC1[*:3]', 'R3': 'O[*:3]'},
 {'Core': 'C1CCC1[*:3]', 'R3': 'F[*:3]'},
 {'Core': 'C1CCC1[*:3]', 'R3': 'C(OC[*:3])[*:3]'}]
```

**Configuration (please complete the following information):**
 - RDKit version: 2020.09.5, master, and master + the changes from #3767

**Comment by jones-gareth** [comment]
@greglandrum I mentioned in #3767 that I had to tweak the fingerprint variance function to encourage it place R groups on labeled R groups.  As implemented in #3767 it doesn't solve this issue.  But if I double the penalty for unoccupied labeled R 
groups then I do get the expected decomp:

![Capture (Custom) (1)](https://user-images.githubusercontent.com/24233741/110423934-2ff89200-805f-11eb-9806-c91ccd4ac771.PNG)

I expect a different decomp from you as the restriction on one heavy substituent per user label means a second r group for the cyclic substituent (or I think it should as in practice each bond to a labeled R group should match, see test `testSDFGRoupMultiCoreNoneShouldMatch` ).

Anyway, It looks like this issue is caused by the scoring function not prioritizing labelled R-groups.  The doubling of the penalty does not affect any of the tests, so I may push this change into #3767.

May make more sense to not allow matches to unlabeled R groups when matches to labeled R groups are found,

**Comment by bp-kelley** [comment]
I think this is resolved, reopen if it hasn't been.
